jstz_api/tests/wpt.rs

```diff
@@ -0,0 +1,326 @@
+    };
+}
+
+impl TestHarnessReportApi {
+    /// The add_result_callback function for jstz's test harness reports
+    ///
+    /// More information:
+    ///  - [wpt documentation][wpt]
+    ///
+    /// [wpt]: https://web-platform-tests.org/writing-tests/testharness-api.html#add_result_callback
+    pub fn test_result_callback(
+        _this: &JsValue,
+        args: &[JsValue],
+        context: &mut Context<'_>,
+    ) -> JsResult<JsValue> {
+        preamble!(context, report);
+
+        let result: TestResult = args.get_or_undefined(0).try_js_into(context)?;
+
+        report.add_test_result(result);
+
+        Ok(JsValue::undefined())
+    }
+
+    pub fn test_completion_callback(
+        _this: &JsValue,
+        args: &[JsValue],
+        context: &mut Context<'_>,
+    ) -> JsResult<JsValue> {
+        preamble!(context, report);
+
+        // The first argument (0) is an array of `Test` objects [ignored]
+        // The second argument (1) is the `TestsStatus` object
+        // The final argument (2) is an array of `AssertRecord` objects [ignored]
+        let result: TestsResult = args.get_or_undefined(1).try_js_into(context)?;
+
+        report.set_harness_result(result)?;
+
+        Ok(JsValue::undefined())
+    }
+}
+
+impl jstz_core::Api for TestHarnessReportApi {
+    fn init(self, context: &mut Context<'_>) {
+        let test_result_callback = FunctionObjectBuilder::new(
+            context.realm(),
+            NativeFunction::from_fn_ptr(Self::test_result_callback),
+        )
+        .name("test_result_callback")
+        .length(1)
+        .build();
+
+        let test_completion_callback = FunctionObjectBuilder::new(
+            context.realm(),
+            NativeFunction::from_fn_ptr(Self::test_completion_callback),
+        )
+        .name("test_completion_callback")
+        .length(3)
+        .build();
+
+        // TODO: Make into a function
+        macro_rules! call_global_function {
+            ($name:literal, $args:expr) => {
+                let value = context
+                    .global_object()
+                    .get(js_string!($name), context)
+                    .expect(&format!("globalThis.{} is undefined", $name));
+
+                let function = value
+                    .as_callable()
+                    .expect(&format!("globalThis.{} is not callable", $name));
+
+                function
+                    .call(&JsValue::undefined(), $args, context)
+                    .expect(&format!("Failed to call globalThis.{}", $name));
+            };
+        }
+
+        call_global_function!("add_result_callback", &[test_result_callback.into()]);
+        call_global_function!(
+            "add_completion_callback",
+            &[test_completion_callback.into()]
+        );
+    }
+}
+
+pub fn run_wpt_test_harness(bundle: &Bundle) -> JsResult<Box<TestHarnessReport>> {
+    let mut rt = Runtime::new()?;
+
+    // Initialize the host-defined object with the test harness report
+    {
```

Comment:
shouldn't we register the API here ?

This worked (i.e. has visible effects on the result)
```
rt.realm()
        .clone()
        .register_api(jstz_api::TextEncoderApi, rt.context());
```

Reply:
Yes. I think perhaps making a function `register_apis` that is passed a context would be best.

Reply:
Added this function ^